[2025-04-26T23:30:40.112+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: earth_monitoring_system.fetch_air_quality_data manual__2025-04-26T23:30:38.052793+00:00 [queued]>
[2025-04-26T23:30:40.123+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: earth_monitoring_system.fetch_air_quality_data manual__2025-04-26T23:30:38.052793+00:00 [queued]>
[2025-04-26T23:30:40.124+0000] {taskinstance.py:1308} INFO - Starting attempt 1 of 2
[2025-04-26T23:30:40.143+0000] {taskinstance.py:1327} INFO - Executing <Task(BashOperator): fetch_air_quality_data> on 2025-04-26 23:30:38.052793+00:00
[2025-04-26T23:30:40.148+0000] {standard_task_runner.py:57} INFO - Started process 204 to run task
[2025-04-26T23:30:40.151+0000] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'earth_monitoring_system', 'fetch_air_quality_data', 'manual__2025-04-26T23:30:38.052793+00:00', '--job-id', '9', '--raw', '--subdir', 'DAGS_FOLDER/earth_monitoring_dag.py', '--cfg-path', '/tmp/tmpxwrifzxk']
[2025-04-26T23:30:40.152+0000] {standard_task_runner.py:85} INFO - Job 9: Subtask fetch_air_quality_data
[2025-04-26T23:30:40.224+0000] {logging_mixin.py:150} WARNING - /home/airflow/.local/lib/python3.7/site-packages/airflow/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-04-26T23:30:40.390+0000] {task_command.py:410} INFO - Running <TaskInstance: earth_monitoring_system.fetch_air_quality_data manual__2025-04-26T23:30:38.052793+00:00 [running]> on host d522fce2db24
[2025-04-26T23:30:40.653+0000] {taskinstance.py:1547} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='earth_monitoring_system' AIRFLOW_CTX_TASK_ID='fetch_air_quality_data' AIRFLOW_CTX_EXECUTION_DATE='2025-04-26T23:30:38.052793+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-26T23:30:38.052793+00:00'
[2025-04-26T23:30:40.654+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-04-26T23:30:40.655+0000] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', 'cd /opt/airflow/data_collection && python fetch_openaq.py']
[2025-04-26T23:30:40.661+0000] {subprocess.py:86} INFO - Output:
[2025-04-26T23:30:41.122+0000] {subprocess.py:93} INFO - 2025-04-26 23:30:41,120 - WARNING - No API key found. Set OPENAQ_API_KEY in .env file.
[2025-04-26T23:30:41.126+0000] {subprocess.py:93} INFO - 2025-04-26 23:30:41,121 - INFO - Starting data collection from OpenAQ v3
[2025-04-26T23:30:41.127+0000] {subprocess.py:93} INFO - 2025-04-26 23:30:41,121 - INFO - Fetching locations from OpenAQ API v3 with limit 1000
[2025-04-26T23:30:41.341+0000] {subprocess.py:93} INFO - 2025-04-26 23:30:41,341 - ERROR - Error fetching data from OpenAQ: 401 Client Error: Unauthorized for url: https://api.openaq.org/v3/locations?limit=1000&has_geo=true
[2025-04-26T23:30:41.342+0000] {subprocess.py:93} INFO - 2025-04-26 23:30:41,342 - WARNING - No data received from API
[2025-04-26T23:30:41.345+0000] {subprocess.py:93} INFO - 2025-04-26 23:30:41,342 - INFO - Data collection completed
[2025-04-26T23:30:41.355+0000] {subprocess.py:97} INFO - Command exited with return code 0
[2025-04-26T23:30:41.453+0000] {taskinstance.py:1350} INFO - Marking task as SUCCESS. dag_id=earth_monitoring_system, task_id=fetch_air_quality_data, execution_date=20250426T233038, start_date=20250426T233040, end_date=20250426T233041
[2025-04-26T23:30:41.523+0000] {local_task_job_runner.py:225} INFO - Task exited with return code 0
[2025-04-26T23:30:41.585+0000] {taskinstance.py:2653} INFO - 0 downstream tasks scheduled from follow-on schedule check
